### PR TITLE
added ANSI color parsing and displaying colors in preview (closes #29)

### LINKED ISSUE #29: Ansi color in preview mode
## Describe the bug
I want to display text with Ansi colors (provided by github.com/logrusorgru/aurora). It's not possible, or it's not make sense?

## To reproduce
```
	projects, err := hbtn.GetProjects()
		if err != nil {
			log.Fatal(err)
		}

		idx, err := fuzzyfinder.FindMulti(
			projects,
			func(i int) string {
				return projects[i].Title
			},
			fuzzyfinder.WithPreviewWindow(func(i, w, h int) string {
				if i == -1 {
					return ""
				}

				return fmt.Sprintf("Track: %s (%s)\nAlbum: %s",
					Green(projects[i].Title),
					projects[i].Tasks,
					projects[i].ID)
			}))
		if err != nil {
			log.Fatal(err)
		}
		fmt.Printf("selected: %v\n", idx)
```
## Screenshots
image

## Environment
- OS: Arch linux
- Terminal: Tilix
- go-fuzzyfinder version: Latest

### PR DESCRIPTION
**Pull request by cynicade**
This is my first PR on Github so feedback is more than welcome.

### CODE REVIEW

**Comment by ktr0731** [comment]
Thanks for your contribution! Since I don't have any time to take a code review, so I'll check this PR later 🙇‍♂️

**Comment by cynicade** [comment]
> Thanks for your contribution! Since I don't have any time to take a code review, so I'll check this PR later bowing_man

No worries, let me know if you'd like me to write tests for this :+1:

File: fuzzyfinder.go [hunk 1]
```diff
@@ -263,6 +264,67 @@ func (f *finder) _draw() {
 	}
 }
 
+func parseColor(rs *[]rune) (tcell.Color, bool, error) {
+	// only parses for 16 colors
+	re, errCompile := regexp.Compile(`\[[0-9;]*m`)
```
Comment (ktr0731) [inline fuzzyfinder.go:269]: Since `Compile` is a slow function, it should be put as a global variable.

Comment (ktr0731) [inline fuzzyfinder.go:269]: Maybe, we can reduce calls of `regexp` functions.

``` go
s := `\[31mfoo`

re := regexp.MustCompile(`(\\\[[0-9;]+m).*`)
subStrs re.FindSubmatch([]byte(s))
if len(subStrs) == 0 {
  // no matching
}

fmt.Println(subStrs[1]) // \[31m
fmt.Println(strings.TrimPrefix(s, string(subStrs[1]))) // foo
fmt.Println(strings.TrimSuffix(strings.TrimPrefix(string(subStrs[1]), `\[`), "m")) // 31
```

regexps for each color can also be replaced with the compare the colors.

``` go
switch n {
case "30":
  return tcell.ColorBlack, ...
case "31":
  return tcell.ColorRed, ...
...
}
```

**Review by ktr0731 — COMMENTED** [review]
Thanks for submitting this PR! I left some comments.

File: fuzzyfinder.go (hunk 1 above)
Comment (cynicade) [reply fuzzyfinder.go]: Thanks for the feedback! I've pushed a new commit with the changes you suggested.

File: fuzzyfinder.go [hunk 2]
```diff
@@ -23,8 +24,9 @@ import (
 
 var (
 	// ErrAbort is returned from Find* functions if there are no selections.
-	ErrAbort   = errors.New("abort")
-	errEntered = errors.New("entered")
+	ErrAbort       = errors.New("abort")
+	errEntered     = errors.New("entered")
+	colorsRegex, _ = regexp.Compile(`\[[0-9;]*m`)
```
Comment (ktr0731) [inline fuzzyfinder.go:29]:
```suggestion
	colorsRegex = regexp.MustCompile(`\[[0-9;]*m`)
```

File: fuzzyfinder.go [hunk 3]
```diff
@@ -263,6 +265,53 @@ func (f *finder) _draw() {
 	}
 }
 
+func parseColor(rs *[]rune) (tcell.Color, bool, error) {
```
Comment (ktr0731) [inline fuzzyfinder.go:268]: Error is never returned 👀 

```suggestion
func parseColor(rs *[]rune) (tcell.Color, bool) {
```

**Review by ktr0731 — COMMENTED** [review]
Thanks for the quick fix! I left some nit comments 🙏

**Review by ktr0731 — APPROVED** [review]
LGTM, thanks for your contribution 👍

**Pull request merged**